site/src/pages/CreateWorkspacePage/CreateWorkspacePageViewExperimental.tsx

```diff
@@ -213,39 +213,76 @@ export const CreateWorkspacePageViewExperimental: FC<
 
 		setPresetParameterNames(selectedPreset.Parameters.map((p) => p.Name));
 
-		for (const presetParameter of selectedPreset.Parameters) {
+		const currentValues = form.values.rich_parameter_values ?? [];
+
+		const updates = selectedPreset.Parameters.map((presetParameter) => {
 			const parameterIndex = parameters.findIndex(
 				(p) => p.name === presetParameter.Name,
 			);
-			if (parameterIndex === -1) continue;
+			if (parameterIndex === -1) return null;
 
 			const parameterField = `rich_parameter_values.${parameterIndex}`;
+			const parameter = parameters[parameterIndex];
+			const currentValue = currentValues.find(
+				(p) => p.name === presetParameter.Name,
+			)?.value;
+
+			if (currentValue !== presetParameter.Value) {
+				return {
+					field: parameterField,
+					fieldValue: {
+						name: presetParameter.Name,
+						value: presetParameter.Value,
+					},
+					parameter,
+					presetValue: presetParameter.Value,
+				};
+			}
+			return null;
+		}).filter(
+			(update): update is NonNullable<typeof update> => update !== null,
+		);
```

Comment:
you could use `flatMap` here instead of `map` and `filter`. just return `[]` or `[x]` instead of `null` and `x`. then we don't need the fancy annotations.

but even that's a bit "clever" for my taste. imho, ideally we'd leave it as a `for` loop and just have some `result` array that I `push` to when desired.